Add SSO configuration endpoints and UI integration with persistent settings

colesmcintosh opened the pull request:
## Title

Add SSO configuration endpoints and UI integration with persistent settings

## Relevant issues

<!-- Please add any relevant issue numbers if applicable -->
<!-- e.g. "Fixes #000" -->

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

## Type

🆕 New Feature
🧹 Refactoring

## Changes

**Problem Solved:**
Previously, users had to re-enter their SSO configuration settings every time they opened the SSO settings modal. The configuration was not being properly saved or loaded, leading to a frustrating user experience where all settings would be lost between sessions.

**Backend Changes:**
- **Persistent SSO Configuration**: Introduced `SSOConfig` class that properly manages SSO settings persistence
- **Data Persistence**: SSO settings are now properly saved to configuration storage and retrieved on subsequent loads
- **Router Integration**: Added new `__init__.py` file to properly expose SSO configuration endpoints

**Frontend Changes:**
- **Configuration Loading**: Enhanced `SSOModals.tsx` to automatically load existing SSO configuration when opened
- **State Persistence**: Users can now:
  - Save their SSO settings and have them persist across browser sessions
  - Return to the SSO configuration modal and see their previously entered values
  - Edit existing configuration without starting from scratch
- **Improved UX**: Added proper loading states while fetching existing configuration
- **Error Handling**: Better feedback when loading or saving configuration fails
- **Networking Integration**: Added new networking functions in `networking.tsx` for reliable SSO config API communication
- **AdminPanel Integration**: Updated `AdminPanel` component to leverage the new persistent SSO system

**Admin Panel UI Improvements:**
- Improved button styles and layout for better accessibility
- Enhanced visual hierarchy and user experience
- Better responsive design patterns that complement the persistent configuration workflow

**User Experience Impact:**

Before this PR:
- Users had to re-enter all SSO configuration settings each time they opened the modal
- No way to review or edit existing SSO settings
- Configuration changes were not reliably saved
- Frustrating workflow that discouraged SSO adoption

After this PR:
- SSO configuration persists across sessions
- Users can return to edit their existing configuration
- Seamless save/load experience
- Encourages proper SSO setup and maintenance

colesmcintosh committed:
Enhance Admin Panel UI: Adjust button styles and layout for better accessibility and user experience

colesmcintosh committed:
Add SSO configuration endpoints and UI integration

- Introduced new SSOConfig class to manage SSO settings.
- Added endpoints for fetching and updating SSO settings in proxy_setting_endpoints.py.
- Created a new __init__.py file to expose the SSO router.
- Updated AdminPanel and SSOModals components to handle SSO settings retrieval and updates.
- Removed deprecated SSO update logic from AdminPanel.
- Enhanced error handling and logging for SSO operations.

colesmcintosh committed:
Remove console log from handleShowInstructions in AdminPanel component

colesmcintosh committed:
Implement SSO configuration check in AdminPanel and update SSOModals to reflect SSO status

- Added logic to check SSO configuration and set state in AdminPanel.
- Introduced a new function to handle SSO configuration checks.
- Updated UI to conditionally render SSO button text based on configuration status.
- Passed SSO configuration status as a prop to SSOModals for better integration.

colesmcintosh committed:
Add clear SSO settings functionality in SSOModals component

- Introduced a confirmation modal for clearing SSO settings.
- Implemented handleClearSSO function to reset SSO settings and provide user feedback.
- Updated UI to include a 'Clear' button for SSO settings, enhancing user experience.
- Added state management for the confirmation modal visibility.

krrishdholakia commented on litellm/proxy/ui_crud_endpoints/proxy_setting_endpoints.py:
```diff
@@ -281,3 +281,128 @@ async def update_default_team_settings(settings: DefaultTeamSSOParams):
+    tags=["SSO Settings"],
+    dependencies=[Depends(user_api_key_auth)],
+)
+async def get_sso_settings():
```
add a response_model to this class, so we are guaranteeing the response type. 

This will prevent any future regressions

colesmcintosh replied:
Updated ✅

colesmcintosh committed:
Add SSO settings response model in proxy_setting_endpoints.py

- Introduced SSOSettingsResponse model to encapsulate SSO configuration values and schema information.
- Updated the get_sso_settings endpoint to utilize the new response model, enhancing API clarity and usability.

colesmcintosh committed:
Refactor settings response models in proxy_setting_endpoints.py

- Renamed SSOSettingsResponse to inherit from a new base class SettingsResponse for better structure.
- Introduced InternalUserSettingsResponse and DefaultTeamSettingsResponse models for internal user and default team settings.
- Updated endpoint responses to use field_schema instead of schema for consistency.
- Enhanced test cases to validate the new response structure and ensure proper functionality of SSO settings.

colesmcintosh committed:
Add uiAuditLogsCall function

colesmcintosh committed:
Merge remote-tracking branch 'origin/main' into sso-config-ui

ishaan-jaff reviewed (approved):
LGTM
